unit-tests/src/main/scala/scala/scalanative/compat/CollectionConverters.scala

```diff
@@ -0,0 +1,61 @@
+package scala.scalanative.compat
+
+import java.util.{LinkedHashMap, LinkedHashSet, LinkedList}
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+/** Set of helper method replacing problematic Scala collection.JavaConverters as they cause problems
+ * in cross compile between 2.13+ and older Scala versions */
+object CollectionConverters {
+  implicit class ScalaToJavaCollections[T: ClassTag](
+      private val self: Iterable[T]) {
+    def toJavaList: LinkedList[T] = {
+      val list = new LinkedList[T]()
+      self.foreach(list.add)
+      list
+    }
+
+    def toJavaSet: java.util.Set[T] = {
+      val s = new LinkedHashSet[T]()
+      self.foreach(s.add)
+      s
+    }
+
+    def toJavaMap[K, V](implicit ev: T =:= (K, V)): java.util.Map[K, V] = {
+      val m = new LinkedHashMap[K, V]()
+      self.iterator.asInstanceOf[Iterator[(K, V)]].foreach {
+        case (k, v) => m.put(k, v)
+      }
```

Comment:
Can we avoid the cast with something like:
```suggestion
      self.foreach { elem =>
        val pair: (K, V) = elem
        m.put(pair._1, pair._2)
      }
```
?